src/main/java/org/spongepowered/common/util/gen/ArrayMutableBlockBuffer.java

```diff
@@ -39,25 +40,28 @@
 import org.spongepowered.common.world.extent.MutableBlockViewTransform;
 import org.spongepowered.common.world.extent.UnmodifiableBlockVolumeWrapper;
 import org.spongepowered.common.world.extent.worker.SpongeMutableBlockVolumeWorker;
+import org.spongepowered.common.world.schematic.BimapPalette;
+import org.spongepowered.common.world.schematic.GlobalPalette;
 
 public class ArrayMutableBlockBuffer extends AbstractBlockBuffer implements MutableBlockVolume {
 
+    private static final int LOCAL_PALETTE_THRESHOLD = 256;
+
     @SuppressWarnings("ConstantConditions")
     private static final BlockState AIR = BlockTypes.AIR.getDefaultState();
 
     private final BlockPalette palette;
     private final BackingData data;
 
-    public ArrayMutableBlockBuffer(BlockPalette palette, Vector3i start, Vector3i size, BackingDataType type) {
-        super(start, size);
-        this.palette = palette;
-        this.data = type.create(size);
+    public ArrayMutableBlockBuffer(Vector3i start, Vector3i size) {
+        this(size.getX() * size.getY() * size.getZ() > LOCAL_PALETTE_THRESHOLD ?
+                new BimapPalette() : GlobalPalette.instance, start, size);
     }
 
-    public ArrayMutableBlockBuffer(BlockPalette palette, Vector3i start, Vector3i size, byte[] blocks) {
+    public ArrayMutableBlockBuffer(BlockPalette palette, Vector3i start, Vector3i size) {
         super(start, size);
         this.palette = palette;
```

Comment:
Likewise here you need to make sure that that we're setting the contents of the backing data to the id for air in the incoming palette (which may not be zero).